pkg/cloudprovider/aws/ami.go

```diff
@@ -83,12 +84,21 @@ func (p *AMIProvider) getAMIID(ctx context.Context, query string) (string, error
 	return ami, nil
 }
 
-func (p *AMIProvider) getSSMQuery(instanceType cloudprovider.InstanceType, version string) string {
+func (p *AMIProvider) getSSMQuery(ctx context.Context, amiFamily string, instanceType cloudprovider.InstanceType, version string) string {
 	var amiSuffix string
+	var arch string
 	if !instanceType.NvidiaGPUs().IsZero() || !instanceType.AWSNeurons().IsZero() {
 		amiSuffix = "-gpu"
 	} else if instanceType.Architecture() == v1alpha5.ArchitectureArm64 {
 		amiSuffix = "-arm64"
+		arch = "arm64"
+	} else if instanceType.Architecture() == v1alpha5.ArchitectureAmd64 {
+		arch = "arm64"
+	}
+
+	if amiFamily == v1alpha5.BottleRocket {
```

Comment:
Thoughts on doing a switch statement here for all ami families? We may need this for flatcar, ubuntu, etc.

Reply:
@ellistarn we actually talked about this today at work.

Is there a concept of "release channels" for these OS versions (bottlerocket and amazon linux)? I see AL2's is called "recommended" - does that imply there's a latest that's newer/less tested? Similarly for bottlerocket - does latest imply there's a recommended?

We were also wondering if it might be beneficial to allow users to specify their own SSM query, so they can control their own "release cadence" outside of AWS' as needed (we might want to do something like let our DEV stuff pull latest, but "pin" PROD to something like "stable" that's more battle tested.